rotkehlchen/api/v1/schemas.py

```diff
@@ -2567,14 +2570,76 @@ def transform_data(
             raise ValidationError(f'Could not deserialize data: {e!s}') from e
 
 
-class LocationAssetMappingsUpdateSchema(LocationAssetMappingsBaseSchema):
+class LocationAssetMappingsUpdateSchema(Schema):
     entries = NonEmptyList(fields.Nested(LocationAssetMappingUpdateEntrySchema), required=True)
 
 
-class LocationAssetMappingsDeleteSchema(LocationAssetMappingsBaseSchema):
+class LocationAssetMappingsDeleteSchema(Schema):
     entries = NonEmptyList(fields.Nested(LocationAssetMappingDeleteEntrySchema), required=True)
 
 
+class CounterpartyAssetMappingsBaseSchema(Schema):
+    counterparty = fields.String(required=True)
```

Comment:
At this point in the API you should check if the counterparty exists. If it does not this should fail at the API level.